rewrite-java/src/main/java/org/openrewrite/java/cleanup/RenameLocalVariablesToCamelCase.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright 2020 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.cleanup;
+
+import org.openrewrite.*;
+import org.openrewrite.internal.lang.Nullable;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.RenameVariable;
+import org.openrewrite.java.tree.J;
+
+import java.util.*;
+import java.util.regex.Pattern;
+
+/**
+ * This recipe converts local variables and method parameters to camel case convention.
+ * The recipe will not rename variables declared in for loop controls or catches with a single character.
+ *
+ * The first character is set to lower case and existing capital letters are preserved.
+ * Special characters that are allowed in java field names `$` and `_` are removed.
+ * If a special character is removed the next valid alpha-numeric will be capitalized.
+ *
+ * Currently, unsupported:
+ *  - The recipe will not rename variables declared in a class.
+ *  - The recipe will not rename variables if the result already exists in a class or the result will be a java reserved keyword.
+ */
+public class RenameLocalVariablesToCamelCase extends Recipe {
+
+    @Override
+    public String getDisplayName() {
+        return "Rename local variables to camel case.";
```

Comment:
Hmm something about the wording here doesn't feel right. Like it's implying that it takes any variable name and makes it "camel case".  Maybe we should talk about "reformatting" the name instead?

```suggestion
        return "Reformat local variable names to camelCase";
```

`getDisplayName()` should not end with a period. See [our guidelines](https://github.com/openrewrite/rewrite/blob/main/doc/adr/0002-recipe-naming.md#display-names) for such names.